numpy/core/src/npysort/quicksort.c.src

```diff
@@ -69,10 +90,12 @@ quicksort_@suff@(void *start, npy_intp num, void *NOT_USED)
     @type@ *stack[PYA_QS_STACK];
     @type@ **sptr = stack;
     @type@ *pm, *pi, *pj, *pk;
-    npy_intp depth_limit = npy_get_msb(num) * 2;
+    int depth[PYA_QS_STACK];
+    int * psdepth = depth;
+    int cdepth = NPY_SORT_MIN(PYA_QS_STACK, npy_get_msb(num) * 2);
```

Comment:
I don't think `* 2` is needed here, just adjust the cutoff point. Then you don't need the `NPY_SORT_MIN` either, especially as it is now `64`. I suppose you could also use `NPY_BITSOF_INTP`